### ISSUE TITLE
Posts federate as links to lemmy instance, as opposed to just the link, since v0.17.0

### ISSUE THREAD

**Issue by roobre** [LemmyNet/lemmy#2776]
Hi!

I have observed that from v0.17.0 onwards, the way post links federate has changed. Prior to v0.17.0, posts viewed outside of lemmy included the link that was being shared directly, while now a link to the post on the lemmy instance is included instead. I think the following screenshots illustrates this:

![image](https://user-images.githubusercontent.com/969721/222683456-3ec8d668-49ad-4ffe-ac8f-9a00d2c7c233.png)

It might be a personal preference, as my Lemmy instance mostly federates with microblogging software, but I pretty much liked the older way better. Is there any way we can get the old behavior back, perhaps through a config setting?

**Comment by Nutomic** [comment]
This must be because of https://github.com/LemmyNet/lemmy/pull/2182. As you can see there the previous behaviour was not standard compliant. If you want Lemmy data to be rendered differently in another platform, you will have to ask the respective developers.

**Comment by roobre** [comment]
Hey @Nutomic,

Thanks a lot for the context! I previously assumed that Lemmy was somehow including the link of the post into the body when federating. It turns out it's not the case! So it would, as you say, be a matter of how other software is rendering the post.

<details>
  <summary>Lemmy post</summary>

```json
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    "https://w3id.org/security/v1",
    {
      "lemmy": "https://join-lemmy.org/ns#",
      "litepub": "http://litepub.social/ns#",
      "pt": "https://joinpeertube.org/ns#",
      "sc": "http://schema.org/",
      "ChatMessage": "litepub:ChatMessage",
      "commentsEnabled": "pt:commentsEnabled",
      "sensitive": "as:sensitive",
      "matrixUserId": "lemmy:matrixUserId",
      "posting_restricted_to_mods": "lemmy:posting_restricted_to_mods",
      "remove_data": "lemmy:remove_data",
      "stickied": "lemmy:stickied",
      "moderators": {
        "@type": "@id",
        "@id": "lemmy:moderators"
      },
      "expires": "as:endTime",
      "distinguished": "lemmy:distinguished",
      "language": "sc:inLanguage"
    }
  ],
  "type": "Page",
  "id": "https://links.roobre.es/post/311",
  "attributedTo": "https://links.roobre.es/u/roobre",
  "to": [
    "https://links.roobre.es/c/tech",
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "name": "You cannot have exactly-once delivery (2015)",
  "cc": [],
  "content": "<p>Source: Hacker News: Newest</p>\n",
  "mediaType": "text/html",
  "source": {
    "content": "Source: Hacker News: Newest",
    "mediaType": "text/markdown"
  },
  "attachment": [
    {
      "href": "https://bravenewgeek.com/you-cannot-have-exactly-once-delivery/",
      "type": "Link"
    }
  ],
  "image": {
    "type": "Image",
    "url": "https://links.roobre.es/pictrs/image/824047ca-3809-444f-b4ab-697c5e368397.jpeg"
  },
  "commentsEnabled": true,
  "sensitive": false,
  "stickied": false,
  "published": "2023-03-03T09:18:29.679093+00:00",
  "audience": "https://links.roobre.es/c/tech"
}

```
</details>


<details>
  <summary>Mastodon post with a link</summary>

```json
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    {
      "ostatus": "http://ostatus.org#",
      "atomUri": "ostatus:atomUri",
      "inReplyToAtomUri": "ostatus:inReplyToAtomUri",
      "conversation": "ostatus:conversation",
      "sensitive": "as:sensitive",
      "toot": "http://joinmastodon.org/ns#",
      "votersCount": "toot:votersCount"
    }
  ],
  "id": "https://mastodon.social/users/agorer/statuses/109959283301030127",
  "type": "Note",
  "summary": null,
  "inReplyTo": null,
  "published": "2023-03-03T12:11:49Z",
  "url": "https://mastodon.social/@agorer/109959283301030127",
  "attributedTo": "https://mastodon.social/users/agorer",
  "to": [
    "https://mastodon.social/users/agorer/followers"
  ],
  "cc": [
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "sensitive": false,
  "atomUri": "https://mastodon.social/users/agorer/statuses/109959283301030127",
  "inReplyToAtomUri": null,
  "conversation": "tag:mastodon.social,2023-03-03:objectId=407878023:objectType=Conversation",
  "content": "<p>Poco hablamos de que las recomendaciones sobre como escribir los avisos de residuos nucleares para futuras civilizaciones son 100% horror lovecraftiano, la verdad</p><p><a href=\"https://en.wikipedia.org/wiki/Long-term_nuclear_waste_warning_messages\" target=\"_blank\" rel=\"nofollow noopener noreferrer\"><span class=\"invisible\">https://</span><span class=\"ellipsis\">en.wikipedia.org/wiki/Long-ter</span><span class=\"invisible\">m_nuclear_waste_warning_messages</span></a></p>",
  "contentMap": {
    "es": "<p>Poco hablamos de que las recomendaciones sobre como escribir los avisos de residuos nucleares para futuras civilizaciones son 100% horror lovecraftiano, la verdad</p><p><a href=\"https://en.wikipedia.org/wiki/Long-term_nuclear_waste_warning_messages\" target=\"_blank\" rel=\"nofollow noopener noreferrer\"><span class=\"invisible\">https://</span><span class=\"ellipsis\">en.wikipedia.org/wiki/Long-ter</span><span class=\"invisible\">m_nuclear_waste_warning_messages</span></a></p>"
  },
  "attachment": [],
  "tag": [],
  "replies": {
    "id": "https://mastodon.social/users/agorer/statuses/109959283301030127/replies",
    "type": "Collection",
    "first": {
      "type": "CollectionPage",
      "next": "https://mastodon.social/users/agorer/statuses/109959283301030127/replies?min_id=109959285348230601&page=true",
      "partOf": "https://mastodon.social/users/agorer/statuses/109959283301030127/replies",
      "items": [
        "https://mastodon.social/users/agorer/statuses/109959285348230601"
      ]
    }
  }
}

```
</details>

It kind of made me curious where Mastodon is getting the link to the lemmy instance then 🤔
From what I can see on the json, only the post ID contains that information. Why would Mastodon arbitrarily decide it should include the post ID in the UI, but not the link attachment?

Perhaps that would be a question for Mastodon though 😅
